### PR TITLE
fix/reset un jail index

### PR DESCRIPTION
**Pull request by sasurobert**
Reset un jail index. Added fix for computing last unjailed bls key from waiting list

### CODE REVIEW

**Review by raduchis — DISMISSED** [review]

File: epochStart/metachain/systemSCs.go [hunk 1]
```diff
@@ -77,6 +79,7 @@ type systemSCProcessor struct {
 	flagSetOwnerEnabled       atomic.Flag
 	flagChangeMaxNodesEnabled atomic.Flag
 	flagStakingV2Enabled      atomic.Flag
+	flagCorrectLastUnjailed   atomic.Flag
```
Comment (SebastianMarian) [inline epochStart/metachain/systemSCs.go:82]: flagCorrectLastUnjailedEnabled

File: epochStart/metachain/systemSCs.go [hunk 2]
```diff
@@ -572,6 +583,34 @@ func (s *systemSCProcessor) updateSystemSCConfigMinNodes() error {
...
+	vmOutput, err := s.systemVM.RunSmartContractCall(vmInput)
+	if err != nil {
+		return err
+	}
+
+	if vmOutput.ReturnCode != vmcommon.Ok {
+		return epochStart.ErrResetLastUnJailedFromQueue
+	}
+
+	err = s.processSCOutputAccounts(vmOutput)
```
Comment (SebastianMarian) [inline epochStart/metachain/systemSCs.go:606]: return s.processSCOutputAccounts(vmOutput) directly

File: epochStart/metachain/systemSCs.go [hunk 3]
```diff
@@ -1245,4 +1284,6 @@ func (s *systemSCProcessor) EpochConfirmed(epoch uint32) {
 		"epoch", epoch,
 		"maxNodes", s.maxNodes,
 	)
+
+	s.flagCorrectLastUnjailed.Toggle(epoch == s.correctLastUnJailEpoch)
```
Comment (SebastianMarian) [inline epochStart/metachain/systemSCs.go:1288]: Add a log for this change

Comment (SebastianMarian) [inline epochStart/metachain/systemSCs.go:1288]: In staking.go this flag is toggled by condition >=, here is ==

Comment (sasurobert) [reply epochStart/metachain/systemSCs.go:1288]: yes. as the reset should happen only once. this is correct.

Comment (sasurobert) [reply epochStart/metachain/systemSCs.go:1288]: done

File: epochStart/metachain/systemSCs.go (hunk 2 above)
Comment (sasurobert) [reply epochStart/metachain/systemSCs.go:606]: would leave like this.

File: epochStart/errors.go [hunk 4]
```diff
@@ -307,3 +307,6 @@ var ErrInvalidTotalToDistribute = errors.New("invalid total to distribute")
 
 // ErrInvalidRewardsPerBlock signals that the computed rewards per block are invalid
 var ErrInvalidRewardsPerBlock = errors.New("invalid rewards per block")
+
+// ErrResetLastUnJailedFromQueue signals that reset unjailed from queue returned with error
+var ErrResetLastUnJailedFromQueue = errors.New("reset last unjailed from queue error")
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply epochStart/errors.go:312]: done

**Review by raduchis — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
Initial system tests passed. 👍

**Pull request merged**